### PR TITLE
GB-609: create/update `.git/gitbutler.json`

### PR DESCRIPTION
**Pull request by extrawurst**
this happens when a new session is created

### CODE REVIEW

File: packages/tauri/src/gb_repository/repository.rs [hunk 1]
```diff
@@ -593,6 +595,28 @@ impl Repository {
...
+        let gb_file_content = serde_json::json!({
+            "sessionId": session_id,
+            "repositoryId": project_id,
+            "gbPath": gb_path,
+        });
+
+        let gb_file_path = project_repository.path().join(".git/gitbutler.json");
+        std::fs::write(&gb_file_path, gb_file_content.to_string())?;
+
+        tracing::info!("gitbutler file updated: {:?}", gb_file_path);
```
Comment (ngalaiko) [inline packages/tauri/src/gb_repository/repository.rs]: let's make it debug, @mtsgrd doesn't like noisy logs 🙂

File: packages/tauri/src/project_repository/repository.rs [hunk 2]
```diff
@@ -99,6 +99,10 @@ impl Repository {
         self.git_repository.path().parent().unwrap()
     }
 
+    pub fn gb_path(&self) -> &std::path::Path {
+        self.git_repository.path()
+    }
```
Comment (ngalaiko) [inline packages/tauri/src/project_repository/repository.rs]: i think what @schacon meant by gb_path is the path of the `gb_repository.git_repository`

anyhow, `gb_` is usually related to our data repository, and here you have a path to the user's project repository that is not gb.

**Review by ngalaiko — CHANGES_REQUESTED** [review]
you have to use a different path, and also maybe add a simple test? we have those in the `gb_repository/repository_tests.rs`

File: packages/tauri/src/project_repository/repository.rs (hunk 2 above)
Comment (schacon) [reply packages/tauri/src/project_repository/repository.rs]: Yes, I wanted the path to the gitbutler git repository, not the project git repository. 👍

**Review by ngalaiko — APPROVED** [review]

**Pull request merged**